### PR TITLE
Refactor Old error types

### PR DESCRIPTION
**Pull request by rleojoseph**
# Description
* Deprecated old error types
* Updated testcases

## Links
[MINI-4544](https://jira.rakuten-it.com/jira/browse/MINI-4544)

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: Sources/Classes/core/Downloader/MiniAppClient.swift [hunk 1]
```diff
@@ -148,39 +148,6 @@ internal class MiniAppClient: NSObject, URLSessionDownloadDelegate {
...
-                    return completionHandler(.failure(failure))
-                }
-                return completionHandler(.success(ResponseData(responseData.data,
-                                                               responseData.httpResponse)))
-            case .failure(let error):
-                MiniAppLogger.d("urlRequest \(urlRequest.url?.absoluteString ?? "-") : Failure", "🔴")
-                return completionHandler(.failure(error))
-            }
-        }
-    }
```
Comment (Climbatize) [inline Sources/Classes/core/Downloader/MiniAppClient.swift:182]: Never noticed we had this in double!

**Review by Climbatize — APPROVED** [review]
LGTM

Maybe we could create :
```swift
typealias MAURLResponseHandler = Result<URL, MASDKError>
typealias MAManifestResponseHandler = Result<ManifestResponse, MASDKError>
typealias MAResponseDataHandler = Result<ResponseData, MASDKError>
```
not sure about the naming, but you get the idea ;)

File: Sources/Classes/core/Downloader/MiniAppClient.swift (hunk 1 above)
Comment (rleojoseph) [reply Sources/Classes/core/Downloader/MiniAppClient.swift:185]: @Climbatize Not sure about other refactorings, but I have used the following,
`typealias MAResponseDataHandler = (Result<ResponseData, MASDKError>) -> Void`
This is used in many places.

**Review by khairul-alam-licon — APPROVED** [review]
Looks good 👍

**Pull request merged**